tests/test_commit.py

```diff
@@ -1,382 +1,382 @@
+    assert repo.git.diff.called is True
+    assert repo.git.commit.called is False
 
 
 def test_commit_no_message(mock_repo):
     """
-    GIVEN GitWrapperCommit initialized with a path and repo
-    WHEN commit is called with no valid message
+    GIVEN GitRepo initialized with a path and repo
+    WHEN commit.commit is called with no valid message
     THEN CommitMessageMissingException raised
     """
-    commit = GitWrapperCommit('./', mock_repo)
+    repo = GitRepo('./', mock_repo)
 
     with pytest.raises(exceptions.CommitMessageMissingException):
-        commit.commit('')
+        repo.commit.commit('')
 
 
 def test_commit_message_is_not_a_string(mock_repo):
     """
-    GIVEN GitWrapperCommit initialized with a path and repo
-    WHEN commit is called with no valid message
+    GIVEN GitRepo initialized with a path and repo
+    WHEN commit.commit is called with no valid message
     THEN CommitMessageMissingException raised
     """
-    commit = GitWrapperCommit('./', mock_repo)
+    repo = GitRepo('./', mock_repo)
 
     with pytest.raises(exceptions.CommitMessageMissingException):
-        commit.commit(True)
+        repo.commit.commit(True)
 
 
 def test_apply_patch(mock_repo):
     """
-    GIVEN GitWrapperCommit initialized with a path and repo
+    GIVEN GitRepo initialized with a path and repo
     WHEN apply_patch is called with a valid branch_name and valid path
     THEN git.am is called
     """
-    commit = GitWrapperCommit('./', mock_repo)
+    commit = GitRepo('./', mock_repo)
 
     with patch('git.repo.fun.name_to_object'):
-        commit.apply_patch('test_branch', './requirements.txt')
+        commit.commit.apply_patch('test_branch', './requirements.txt')
     assert commit.repo.git.am.called is True
 
 
 def test_apply_patch_wrong_branch_name(mock_repo):
     """
-    GIVEN GitWrapperCommit initialized with a path and repo
+    GIVEN GitRepo initialized with a path and repo
     WHEN apply_patch is called with an invalid branch_name and valid path
     THEN ReferenceNotFoundExceptionRaised
     AND git.am not called
     """
-    commit = GitWrapperCommit('./', mock_repo)
+    commit = GitRepo('./', mock_repo)
 
     with patch('git.repo.fun.name_to_object') as mock_name_to_object:
         mock_name_to_object.side_effect = git.exc.BadName()
         with pytest.raises(exceptions.ReferenceNotFoundException):
-            commit.apply_patch('invalid_branch', './requirements.txt')
+            commit.commit.apply_patch('invalid_branch', './requirements.txt')
     assert commit.repo.git.am.called is False
 
 
 def test_apply_patch_not_a_file(mock_repo):
     """
-    GIVEN GitWrapperCommit initialized with a path and repo
+    GIVEN GitRepo initialized with a path and repo
     WHEN apply_patch is called with a valid branch_name and invalid path
     THEN FileDoesntExistException raised
     AND git.am not called
     """
-    commit = GitWrapperCommit('./', mock_repo)
+    commit = GitRepo('./', mock_repo)
```

Comment:
Minor thought here that did not occur to me in earlier reviews.  What if you made this the actual commit object contained in the GitRepo object as you had before?  Then you could simply call commit.action below instead of commit,commit.action. Maybe not worth it, and certainly not required for this patchset, just a thought - I am guessing there is some reason you switched, and I am just not getting it on a Friday afternoon...

Reply:
My thoughts are that from an end-user perspective it makes more sense to have your generic repo object and call what you need as you need it, rather than expect in advance you'll only need a subset, and be stuck when actually you need to do something else too. I changed the tests to reflect this usage pattern as well.

Having said that commit.commit() is the naming I'm least happy with, but that's easy to deprecate and change later if we decide commit.create() or something else is more sensible.